src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -0,0 +1,641 @@
+		public enum IncrementDirection
+		{
+			Lower,
+			Higher,
+		};
+
+		public enum IncrementAmount
+		{
+			Small,
+			Large,
+		};
+
+		public static Hsv IncrementColorChannel(
+			Hsv originalHsv,
+			ColorPickerHsvChannel channel,
+			IncrementDirection direction,
+			IncrementAmount amount,
+			bool shouldWrap,
+			double minBound,
+			double maxBound)
+		{
+			Hsv newHsv = originalHsv;
+
+			if (amount == IncrementAmount.Small || !DownlevelHelper.ToDisplayNameExists())
+			{
+				// In order to avoid working with small values that can incur rounding issues,
+				// we'll multiple saturation and value by 100 to put them in the range of 0-100 instead of 0-1.
+				newHsv.S *= 100;
+				newHsv.V *= 100;
+
+				// Uno Doc: *valueToIncrement removed for C#
+				double valueToIncrement = 0.0;
+				double incrementAmount = 0.0;
+
+				// If we're adding a small increment, then we'll just add or subtract 1.
+				// If we're adding a large increment, then we want to snap to the next
+				// or previous major value - for hue, this is every increment of 30;
+				// for saturation and value, this is every increment of 10.
+				switch (channel)
+				{
+					case ColorPickerHsvChannel.Hue:
+						valueToIncrement = newHsv.H;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 30;
+						break;
+
+					case ColorPickerHsvChannel.Saturation:
+						valueToIncrement = newHsv.S;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 10;
+						break;
+
+					case ColorPickerHsvChannel.Value:
+						valueToIncrement = newHsv.V;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 10;
+						break;
+
+					default:
+						throw new Exception(); // Uno Doc: 'winrt::hresult_error(E_FAIL);'
+				}
+
+				double previousValue = valueToIncrement;
+
+				valueToIncrement += (direction == IncrementDirection.Lower ? -incrementAmount : incrementAmount);
+
+				// If the value has reached outside the bounds, we were previous at the boundary, and we should wrap,
+				// then we'll place the selection on the other side of the spectrum.
+				// Otherwise, we'll place it on the boundary that was exceeded.
+				if (valueToIncrement < minBound)
+				{
+					valueToIncrement = (shouldWrap && previousValue == minBound) ? maxBound : minBound;
+				}
+
+				if (valueToIncrement > maxBound)
+				{
+					valueToIncrement = (shouldWrap && previousValue == maxBound) ? minBound : maxBound;
+				}
+
+				// Uno Doc: WinUI/C++ modifies the newHsv channel directly using a pointer to the appropriate channel.
+				// This isn't desirable in C# so once the modification is done it has to be copied back to newHsv.
+				switch (channel)
+				{
+					case ColorPickerHsvChannel.Hue:
+						newHsv.H = valueToIncrement;
+						break;
+
+					case ColorPickerHsvChannel.Saturation:
+						newHsv.S = valueToIncrement;
+						break;
+
+					case ColorPickerHsvChannel.Value:
+						newHsv.V = valueToIncrement;
+						break;
+
+					default:
+						throw new Exception();
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)